dlt/dlt_broker.py

```diff
@@ -82,17 +83,33 @@ def __init__(
             self.filter_ack_queue = None
             self.filter_ack_msg_handler = None
 
-        kwargs["ip_address"] = ip_address
-        kwargs["port"] = port
-        kwargs["timeout"] = kwargs.get("timeout", DLT_CLIENT_TIMEOUT)
-        self.msg_handler = DLTMessageHandler(
-            self.filter_queue,
-            self.message_queue,
-            self.mp_stop_flag,
-            kwargs,
-            dlt_time_value=self._dlt_time_value,
-            filter_ack_queue=self.filter_ack_queue,
-        )
+        self._from_dlt_client = True if ip_address else False
+
+        if self._from_dlt_client:
```

Comment:
I don't really like this way to initialize an object inside another class. What if we want to another DLTMessageHandler in the feature? add more else if? It also causes the function signature of `__init__` is not easy to understand and maintain.

Please think about how to initialize DLTBroker with external objects.

Hint: Dependency Injection.

Reply:
I was also thinking about it:

class DLTBroker(...dlt_message_handler...):
     self.msg_handler = dlt_message_handler

but this could cause modification to DLTConnector as well:

class DLTConnector(Connector):
    def  start(self, target):
         dlt_message = DLTMessageHandler(...)
         self.broker = DLTBroker(dlt_message )
         
Not only this part, but also the commonality between DLTFileSpinner and DLTMessageHandler, which only have different run(). A common base class is easier for maintainence. But that would also touch the code of DLTConnector part.

I would prefer not to touch anything to DLTConnector at this PR, but with a following ticket.

Reply:
https://realpython.com/python-multiple-constructors/#providing-multiple-constructors-with-classmethod-in-python

Reply:
I would go against modifying DLTConnector (or whatever else is using DLTBroker). DLTBroker has its own responsibilities of initializing and keeping up the connection to the DLT data source using the parameters given to it. It's perfectly fine for it to choos what data source to use.

Reply:
But for two classes that have most of the methods being identical, well - why not simply inherit one from the other and override one method? There is no need to make an abstract common class if there is no complex inheritance structure, it is perfectly fine to just have a functional class and then another functional class that is "the same, but with a small twist" which inherits from first

Reply:
Aigars, I'm holding at such ugly state with duplicate code, because working on the base class could touch the part for DLTConnector, because DLTMessageHandler is singly used by DLTConnector.

And dltConnector will get fixed later with https://asc.bmwgroup.net/mgujira/browse/SIT-39098, which could result in a better base class.

so, my plan is:
1. first update: only singly add another functionality for ioc dlt connector, withouting touching the hu dlt connector at all
2. fix https://asc.bmwgroup.net/mgujira/browse/SIT-39098 and work on the common base class.

So, if the behaviour of DLTConnector is unexpectedly changed, then out of the worst case, we just need to revoke the 2nd step.

Reply:
@lzhou3957 Please remove internal link. It's a public repo.

Reply:
If you really wan to refactor it later. it's ok. Can you show your plan how to change the code?

For me, inheritance is not a good solution for this problem. 

Please refer this article (it's highly recommended to read) and this sentence.
https://hynek.me/articles/python-subclassing-redux/
> [For inheritance] In all cases you sacrifice reading convenience for writing convenience.

Reply:
thanks for your acceptance of leaving it in such mess way. I know it is not easy for you.

but the code refactory is another topic, that means, disagreement on the way of code refactory should not block the merge of this code.

to be honest, i think a common base class is enough for this easy scenario. thanks for sharing the link, but i dont see extra big benefits of using that for this easy case.

Reply:
I don't say that I accept your plan